pkg/controllers/static/deprovisioning/controller.go

```diff
@@ -174,11 +173,72 @@ func HasNodePoolReplicaCountChanged(oldNP, newNP *v1.NodePool) bool {
-// 1. Empty nodes (nodes with no pods or only DaemonSet pods without do-not-disrupt annotation)
-// 2. If more nodes needed, nodes with lowest disruption cost (nodes with pods that have do-not-disrupt will have highest cost)
-func (c *Controller) getDeprovisioningCandidates(ctx context.Context, np *v1.NodePool, nodes []*state.StateNode, count int) []*state.StateNode {
-	// First get empty nodes
+// Returns NodeClaims suitable for deprovisioning, prioritizing:
+// 1. Unresolved NodeClaims (no ProviderID yet - haven't launched)
+// 2. Empty nodes (nodes with no pods or only DaemonSet pods without do-not-disrupt annotation)
+// 3. If more nodes needed, nodes with lowest disruption cost (nodes with pods that have do-not-disrupt will have highest cost)
+func (c *Controller) getDeprovisioningCandidates(ctx context.Context, np *v1.NodePool, count int) []*v1.NodeClaim {
+	candidates := make([]*v1.NodeClaim, 0, count)
+
+	// Unresolved NodeClaims (haven't launched yet or that failed CreateFleet call)
+	unresolvedCandidates := c.getUnresolvedNodeClaims(ctx, np.Name, count)
+	candidates = append(candidates, unresolvedCandidates...)
+	remaining := count - len(candidates)
+
+	if remaining == 0 {
+		return candidates
+	}
+
+	// Get all StateNodes for this NodePool
+	nodes := make([]*state.StateNode, 0)
+	for n := range c.cluster.Nodes() {
+		if n.Labels()[v1.NodePoolLabelKey] == np.Name && n.NodeClaim != nil && !n.MarkedForDeletion() {
+			nodes = append(nodes, n.DeepCopy())
+		}
+	}
+
+	// Resolved nodes (empty first, then by disruption cost)
+	resolvedCandidates := c.getResolvedNodeClaims(ctx, nodes, np, remaining)
+	candidates = append(candidates, resolvedCandidates...)
+
+	return candidates
+}
+
+// getUnresolvedNodeClaims returns unresolved NodeClaims (those without ProviderID) up to the specified count
+func (c *Controller) getUnresolvedNodeClaims(ctx context.Context, nodePoolName string, count int) []*v1.NodeClaim {
+	nodeClaimList := &v1.NodeClaimList{}
+	if err := c.kubeClient.List(ctx, nodeClaimList, client.MatchingLabels{v1.NodePoolLabelKey: nodePoolName}); err != nil {
+		log.FromContext(ctx).Error(err, "failed to list nodeclaims")
+		return nil
+	}
+
+	unresolvedNodeClaims := lo.Filter(nodeClaimList.Items, func(nc v1.NodeClaim, _ int) bool {
+		return nc.Status.ProviderID == "" && nc.DeletionTimestamp.IsZero()
+	})
+
+	if len(unresolvedNodeClaims) == 0 {
+		return nil
+	}
+
+	unresolvedToDelete := lo.Min([]int{count, len(unresolvedNodeClaims)})
+	candidates := make([]*v1.NodeClaim, 0, unresolvedToDelete)
+	for i := range unresolvedToDelete {
+		candidates = append(candidates, &unresolvedNodeClaims[i])
+	}
+
+	return candidates
+}
+
+// getResolvedNodeClaims returns resolved NodeClaims (those with ProviderID) up to the specified count,
+// prioritizing empty nodes first, then nodes with lowest disruption cost
+func (c *Controller) getResolvedNodeClaims(ctx context.Context, nodes []*state.StateNode, np *v1.NodePool, count int) []*v1.NodeClaim {
```

Comment:
Similar naming comment to my previous one
```suggestion
func (c *Controller) resolvedDeprovisioningCandidates(ctx context.Context, nodes []*state.StateNode, np *v1.NodePool, count int) []*v1.NodeClaim {
```